### DISCUSSION TITLE
What is the difference between RegisterRowMapper and UseRowMapper ?

### DISCUSSION THREAD

**Discussion by fedxyz** [jdbi/jdbi#2581] [Q&A]
Hi,

question in the title.

Thanks for the help

**Comment by stevenschlansker** [comment]
Hi, RegisterRowMapper establishes a mapping from a Java type `T` to a mapper factory `F`, such that when Jdbi is asked to produce a type `T` and doesn't otherwise know what to do, it looks up `F` and uses it.

UseRowMapper is more direct and says, use this row mapper exactly to produce the result, without doing any register or lookup based on the type.
